invenio_app_rdm/records_ui/views/deposits.py

```diff
@@ -78,145 +78,186 @@ def get_form_pids_config():
-    )
-
-
-def _dump_subjects_vocabulary():
-    """Dump subjects vocabulary (limitTo really)."""
-    subjects = (
-        VocabularyScheme.query
-        .filter_by(parent_id="subjects")
-        .options(load_only("id"))
-        .all()
-    )
-    limit_to = [{"text": "All", "value": "all"}]
-    # id is human readable and shorter, so we use it
-    limit_to += [{"text": s.id, "value": s.id} for s in subjects]
-    return {
-        "limit_to": limit_to
-    }
-
-
-def _dump_vocabulary_w_basic_fields(vocabulary_type):
-    """Dump vocabulary with id and title field."""
-    results = vocabulary_service.read_all(
-        system_identity, fields=["id", "title"], type=vocabulary_type)
-    return [
-        {
-            "text": gettext_from_dict(
-                r["title"], current_i18n.locale,
-                current_app.config.get('BABEL_DEFAULT_LOCALE', 'en')
-            ),
-            "value": r["id"],
-        } for r in results.to_dict()["hits"]["hits"]
-    ]
-
-
-def _dump_title_types_vocabulary():
-    """Dump title type vocabulary."""
-    return _dump_vocabulary_w_basic_fields('titletypes')
-
-
-def _dump_creators_role_vocabulary():
-    """Dump creators role vocabulary."""
-    return _dump_vocabulary_w_basic_fields('creatorsroles')
-
-
-def _dump_contributors_role_vocabulary():
-    """Dump contributors role vocabulary."""
-    return _dump_vocabulary_w_basic_fields('contributorsroles')
-
-
-def _dump_description_types_vocabulary():
-    """Dump description type vocabulary."""
-    return _dump_vocabulary_w_basic_fields('descriptiontypes')
-
-
-def _dump_date_types_vocabulary():
-    """Dump date type vocabulary."""
-    return _dump_vocabulary_w_basic_fields('datetypes')
-
-
-def _dump_relation_types_vocabulary():
-    """Dump relation type vocabulary."""
-    return _dump_vocabulary_w_basic_fields('relationtypes')
-
-
-def _dump_identifiers_scheme_label():
-    """Dump identifiers scheme (fake) vocabulary.
+class VocabulariesOptions:
+    """Holds React form vocabularies options."""
+
+    def __init__(self):
+        """Constructor."""
+        self._vocabularies = {}
+
+    # Utilities
+    def _get_label(self, hit):
+        """Return label (translated title) of hit."""
+        return gettext_from_dict(
+            hit["title"],
+            current_i18n.locale,
+            current_app.config.get('BABEL_DEFAULT_LOCALE', 'en')
+        )
+
+    def _get_type_subtype_label(self, hit, type_labels):
+        """Return (type, subtype) pair for this hit."""
+        id_ = hit["id"]
+        type_ = hit.get("props", {}).get("type")
+
+        if id_ == type_:
+            # dataset-like case
+            return (self._get_label(hit), "")
+        elif type_ not in type_labels:
+            # safety net to generate a valid type, subtype and not break search
+            return (self._get_label(hit), "")
+        else:
+            return (type_labels[type_], self._get_label(hit))
```

Comment:
This seems a bit complex/specific for our resource type values, could we simplify it to this?
```suggestion
    def _get_type_subtype_label(self, hit):
        """Return (type, subtype) pair for this hit."""
        type_name = hit.get("props", {}).get("type_name")
        subtype_name = hit.get("props", {}).get("subtype_name")
        return type_name, subtype_name
```

Reply:
Ah, those (`type_name` and `subtype_name`) are the fields we are trying not to use anymore, because they are not translations. We are trying to rely on `title` only and eventually we can phase them out.